schema_editor/test/spec/schemas/schemas-service.spec.js

```diff
@@ -148,4 +176,69 @@ describe('ase.schemas:Schemas', function () {
             message: jasmine.stringMatching('Diff')
         }]));
     });
+
+    // NOTE: This test must be updated when new field types are added.
+    it('should preserve all information when serializing and deserializing schemas', function () {
+        // In practice this means that the serialize and deserialize methods are inverses
```

Comment:
Ok. I don't have any offhand ideas due to relative unfamiliarity with some of the newer stuff here. Maybe inspiration will strike once I dig in again.